fr.esrf.tango.generator.xtend/src/fr/esrf/tango/pogo/generator/python/PythonDeviceHL.xtend

```diff
@@ -212,13 +214,31 @@ from tango.server import Device, DeviceMeta
 «IF !cls.classProperties.empty || !cls.deviceProperties.empty»from tango.server import «IF !cls.classProperties.empty»class_property, «ENDIF»«IF !cls.deviceProperties.empty»device_property«ENDIF»«ENDIF»
 from tango import AttrQuality, DispLevel, DevState
 from tango import AttrWriteType, PipeWriteType
+from enum import IntEnum
 «cls.inheritedAdditionalImportHL»
 # Additional import
 «IF cls.description.filestogenerate.toLowerCase.contains("protected regions")»«cls.protectedAreaHL("additionnal_import")»«ENDIF»
 
 __all__ = ["«cls.name»", "main"]
 '''
 
+def enumClasses(PogoDeviceClass cls) '''
+«IF cls.attributes!==null»
+«FOR attr:cls.attributes»
+«IF attr.dataType.pythonTypeHL.equalsIgnoreCase("'DevEnum'")»
+
+
+class «attr.name.toFirstUpper»(IntEnum):
+    """«attr.name.toFirstUpper» enum."""
+    «IF attr.enumLabels!==null»
+    «IF attr.enumLabels.size>0»
+    «attr.enumLabelsWithNumber»

```

Comment:
This is fine for enums that use labels that are valid identifiers, but won't work with a label like `MY-STATE`.  SKA code has some like that:  https://gitlab.com/ska-telescope/lmc-base-classes/blob/b3cc537e/skabase/SKABaseDevice/SKABaseDevice.xmi#L104

The suggestion in issue #80 is to use the functional API instead.  I think it would be nicest if the code generation uses the "normal" `class «attr.name.toFirstUpper»(IntEnum):` form if all the labels are valid identifiers, and falls back to the functional API only if an enum has invalid labels.  However, it may be difficult to determine "valid" Python identifier in Java code.

Reply:
I got your point. As '-' is not a valid python identifier, Enum labels generated with '-' won't be accepted. The suggestion in #80 provides a way to access such Enum labels. 
But won't there be inconsistency in accessing Enum labels as it depends on the label value? Instead can we change Enum labels with underscore instead of hyphen? SKA Control Model also does not have hyphen in Enum labels as per [SKA Control Model - Draft 3](https://docs.google.com/document/d/1422IecZUcjBelwITCGqc3Byop9FzD5UHd36CD-hy6qw/edit#). 
Earlier underscore was not supported for Enum labels, but from PR #64, underscore is now allowed. Please let me know your views on this.

Reply:
I would like the SKA enum labels to be changed to underscores,.  Independent of that, the code generated by Pogo must not have syntax errors, for valid TANGO labels.

Reply:
Okay Thanks. I will push the code with the changes.